src/main/java/io/lettuce/core/cluster/PubSubClusterEndpoint.java

```diff
@@ -189,6 +196,20 @@ public void punsubscribed(RedisClusterNode node, K pattern, long count) {
             clusterListeners.forEach(listener -> listener.punsubscribed(node, pattern, count));
         }
 
+        @Override
+        public void ssubscribed(RedisClusterNode node, K channel, long count) {
+            getListeners().forEach(listener -> {
+                if (listener instanceof RedisShardedPubSubListener) {
```

Comment:
We should avoid using `instanceof` to implement different implementations. See note on this topic below.